### ISSUE TITLE
[CI] Build Tools Unit Tests Failed on 6.8 (JDK14)

### ISSUE THREAD

**Issue by original-brownbear** [elastic/elasticsearch#46018]
See https://gradle-enterprise.elastic.co/s/6bsxf53wac2oo/console-log

The run failed with a bunch of errors like:

```

:build-tools:unitTest FAILED
--
==> Test Info: seed=568609D4AE6A85C7; jvms=7; suites=7
Suite: org.elasticsearch.gradle.doc.RestTestFromSnippetsTaskTests
2> java.lang.NoClassDefFoundError: Could not initialize class org.codehaus.groovy.vmplugin.v7.Java7
2> 	at org.codehaus.groovy.vmplugin.VMPluginFactory.<clinit>(VMPluginFactory.java:43)
2> 	at org.codehaus.groovy.reflection.GroovyClassValueFactory.<clinit>(GroovyClassValueFactory.java:35)
2> 	at org.codehaus.groovy.reflection.ClassInfo.<clinit>(ClassInfo.java:109)
2> 	at org.codehaus.groovy.reflection.ReflectionCache.getCachedClass(ReflectionCache.java:95)
2> 	at org.codehaus.groovy.reflection.ReflectionCache.<clinit>(ReflectionCache.java:39)
2> 	at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.registerMethods(MetaClassRegistryImpl.java:209)
2> 	at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.<init>(MetaClassRegistryImpl.java:107)
2> 	at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.<init>(MetaClassRegistryImpl.java:85)
2> 	at groovy.lang.GroovySystem.<clinit>(GroovySystem.java:36)
2> 	at org.codehaus.groovy.runtime.InvokerHelper.<clinit>(InvokerHelper.java:86)
2> 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.createCallStaticSite(CallSiteArray.java:74)
2> 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.createCallSite(CallSiteArray.java:161)
2> 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
2> 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115)
2> 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:119)
2> 	at org.elasticsearch.gradle.doc.RestTestFromSnippetsTaskTests.<init>(RestTestFromSnippetsTaskTests.groovy:33)
2> 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
2> 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
2> 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
2> 	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
2> 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
2> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$NewInstanceProvider.newInstance(RandomizedRunner.java:1580)
2> 	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:933)
2> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
2> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
2> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
2> 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
2> 	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
2> 	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
2> 	at com.carrotsearch.randomizedtesting.ThreadLeakControl$2.evaluate(ThreadLeakControl.java:404)
2> 	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSuite(RandomizedRunner.java:708)
2> 	at com.carrotsearch.randomizedtesting.RandomizedRunner.access$200(RandomizedRunner.java:138)
2> 	at com.carrotsearch.randomizedtesting.RandomizedRunner$2.run(RandomizedRunner.java:629)
ERROR   0.16s J3 \| RestTestFromSnippetsTaskTests.testSimpleBlockQuote <<< FAILURES!
[ant:junit4] JVM J1: stderr was not empty, see: /var/lib/jenkins/workspace/elastic+elasticsearch+6.8+matrix-java-periodic/ES_BUILD_JAVA/openjdk12/ES_RUNTIME_JAVA/openjdk14/nodes/general-purpose/buildSrc/build/testrun/unitTest/temp/junit4-J1-20190826_235754_5502325830710786880065.syserr
> Throwable #1: java.lang.NoClassDefFoundError: Could not initialize class org.codehaus.groovy.reflection.ReflectionCache


```

**Comment by matriv** [comment]
Another failure: Log: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.8+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=openjdk14,nodes=general-purpose/172/console Build scan: https://gradle-enterprise.elastic.co/s/iao4hxqaujx62

**Comment by rjernst** [comment]
This is a known gradle issue: https://github.com/gradle/gradle/issues/10248

I wonder if we should disable jdk 14 tests until it is fixed. It will continue failing until we have a new version of gradle that supports it.

**Comment by rjernst** [comment]
Upon further investigation, this is only an issue because the build tools tests are using the runtime java instead of the build java. We should override the java used in build-tools to be the same as the gradle jvm.

**Comment by rjernst** [comment]
This was fixed by #46228.
